File: 04_control_flujo_ejecucion/ejemplos/programa_corregido.py
"""
Versión corregida de programa_con_bugs.py.

⚠️ No mires este archivo hasta haber intentado encontrar y arreglar
el bug tú mismo. Depurar es lo que se aprende.

Tema 4 - Introducción a la Computación Científica (ICC).
"""

n = 5

suma = 0
i = 0
while i < n:
    i = i + 1
    suma = suma + 2 * i   # ✅ ya solo incrementamos una vez

print(f"Suma de los {n} primeros pares: {suma}")
print("Esperado: 30")

File: 04_control_flujo_ejecucion/ejemplos/programa_corregido.c
/*
 * Versión corregida de programa_con_bugs.c.
 *
 * ⚠️ No mires este archivo hasta haber intentado encontrar y arreglar
 * el bug tú mismo. Depurar es lo que se aprende.
 *
 * Compilar con: gcc -o programa programa_corregido.c
 *
 * Tema 4 - Introducción a la Computación Científica (ICC).
 */

#include <stdio.h>

int main(void) {
    int n = 5;
    int suma = 0;
    int i = 0;

    while (i < n) {
        i = i + 1;
        suma = suma + 2 * i;   /* ✅ ya solo incrementamos una vez */
    }

    printf("Suma de los %d primeros pares: %d\n", n, suma);
    printf("Esperado: 30\n");

    return 0;
}
